core/src/test/scala/unit/kafka/log/LogSegmentTest.scala

```diff
@@ -65,6 +66,21 @@ class LogSegmentTest {
     Utils.delete(logDir)
   }
 
+  /**
+   * If the maximum offset beyond index, appended to the log section, it throws LogSegmentOffsetOverflowException
+   */
+  @Test
+  def testAppendForLogSegmentOffsetOverflowException(): Unit = {
+    val seg = createSegment(0)
+    val largestOffset: Long = Integer.MAX_VALUE + 1
```

Comment:
I would suggest to use `@ParameterizedTest` and also test for:
1. Negative value for largestOffset
2. Value exceeding the range of integer (existing)

Reply:
Good idea